dojo/tools/bugcrowd_api/parser.py

```diff
@@ -0,0 +1,227 @@
+import json
+import textwrap
+from datetime import datetime
+from dojo.models import Endpoint, Finding
+from dojo.tools.bugcrowd_api.importer import BugcrowdApiImporter
+import re
+import dateutil.parser
+import logging
+
+SCAN_BUGCROWD_API = "Bugcrowd API Import"
+
+pattern_title_authorized = re.compile(r"^[a-zA-Z0-9_\s+-.]*$")
+
+logger = logging.getLogger(__name__)
+
+
+class BugcrowdApiParser(object):
+    """
+    Import from Bugcrowd API /submissions
+    """
+
+    def get_scan_types(self):
+        return [SCAN_BUGCROWD_API]
+
+    def get_label_for_scan_types(self, scan_type):
+        return SCAN_BUGCROWD_API
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Bugcrowd submissions can be directly imported using the Bugcrowd API. An API Scan Configuration has to be setup in the Product."
+
+    def requires_file(self, scan_type):
+        return False
+
+    def requires_tool_type(self, scan_type):
+        return "Bugcrowd API"
+
+    def get_findings(self, file, test):
+        if file is None:
+            data = BugcrowdApiImporter().get_findings(test)
+        else:
+            data = json.load(file)
+        findings = []
+
+        for entry in data:
+            if test.api_scan_configuration:
+                config = test.api_scan_configuration
+                links = "https://tracker.bugcrowd.com/{}{}".format(
+                    str(config.service_key_1), entry["links"]["self"]
+                )
+            else:
+                if "links" in entry:
+                    if "self" in entry["links"]:
+                        links = entry["links"]["self"]
+
+            if not self.include_finding(entry):
+                continue
+
+            bugcrowd_state = entry["attributes"]["state"]
+            bugcrowd_duplicate = entry["attributes"]["duplicate"]
+            bugcrowd_severity = entry["attributes"]["severity"]
+
+            title = entry["attributes"]["title"]
+
+            if not pattern_title_authorized.match(title):
+                char_to_replace = {":": " ", '"': " ", "@": "at"}
+                for key, value in char_to_replace.items():
+                    title = title.replace(key, value)
+
+            date = dateutil.parser.parse(entry["attributes"]["submitted_at"])
+
+            bug_url = ""
+            bug_endpoint = Endpoint()
```

Comment:
Empty `Endpoint` like `Endpoint()` will raise an exception, when you call `.clean()`
https://github.com/DefectDojo/django-DefectDojo/blob/dc20ee1bf9f99953b50c610bd9840053e8fd12f1/dojo/models.py#L1442-L1449
I suppose, you should set `bug_endpoint`  to `None`

And this part (later in the code):
```python
    try:
        bug_endpoint.clean()
        finding.unsaved_endpoints = [bug_endpoint]
```
should be called after
```python
    if bug_endpoint:
```